engine/client_options.go

```diff
@@ -579,46 +532,21 @@ func WithNotifications() ClientOps {
 	}
 }
 
-// WithFeeQuotes will find the lowest fee instead of using the fee set by the WithFeeUnit function
-func WithFeeQuotes(enabled bool) ClientOps {
-	return func(c *clientOptions) {
-		c.chainstate.options = append(c.chainstate.options, chainstate.WithFeeQuotes(enabled))
-	}
-}
-
-// WithFeeUnit will set the fee unit to use for broadcasting
-func WithFeeUnit(feeUnit *bsv.FeeUnit) ClientOps {
-	return func(c *clientOptions) {
-		c.chainstate.options = append(c.chainstate.options, chainstate.WithFeeUnit(feeUnit))
-	}
-}
-
-// WithBroadcastClient will set broadcast client
-func WithBroadcastClient(broadcastClient broadcast.Client) ClientOps {
-	return func(c *clientOptions) {
-		c.chainstate.options = append(c.chainstate.options, chainstate.WithBroadcastClient(broadcastClient))
-	}
-}
+// -----------------------------------------------------------------
+// CHAIN
+// -----------------------------------------------------------------
 
-// WithCallback set callback settings
-func WithCallback(callbackURL string, callbackToken string) ClientOps {
+// WithCustomFeeUnit will set the custom fee unit for transactions
+func WithCustomFeeUnit(feeUnit bsv.FeeUnit) ClientOps {
 	return func(c *clientOptions) {
-		c.txCallbackConfig = &txCallbackConfig{
-			URL:   callbackURL,
-			Token: callbackToken,
-		}
-		c.chainstate.options = append(c.chainstate.options, chainstate.WithCallback(callbackURL, callbackToken))
+		c.feeUnit = &feeUnit
 	}
 }
 
-// WithARC set ARC url params
-func WithARC(url, token, deploymentID string) ClientOps {
+// WithARC sets all the ARC options needed for broadcasting, querying transactions etc.
+func WithARC(arcCfg chainmodels.ARCConfig) ClientOps {
```

Comment:
are we making any validation of this config?

Reply:
At config level, yes.
https://github.com/bitcoin-sv/spv-wallet/blob/c25c6f8c7bdfe832e7908502ba4887158a26b132/config/validate_arc.go#L16